docs/literate/make.jl

```diff
@@ -75,7 +75,16 @@ function create_tutorials(files)
     end
 
     # Generate markdown file for introduction page
-    Literate.markdown(joinpath(repo_src, "index.jl"), pages_dir; name="introduction")
+    # Preprocessing introduction file: Generate tutorial numbers
+    function preprocess_introduction(content)
+        counter = 1
+        while occursin("\$index", content)
+            content = replace(content, "\$index" => "$counter", count = 1)
```

Comment:
Not sure if we want to use `$index`, since it looks like a standard Julia variable interpolation setup. Since this is some very special sauce we're adding here, maybe use something that immediately looks like a template variable but does not suggest it's a Julia-internal feature, e.g., `{index}`? This is just a suggestion though

Reply:
Then, I'd also change the note for developers to `"<Left Curly Brace>index<Right Curly Brace>"`. Or do you think something else is more understandable?

Reply:
You could also use something like
```md
#src Note to developers: Use "{ index }" (but without spaces, see next line) to enable automatic indexing
```
or similar, but yours is also fine

Reply:
Done in [b88f128](https://github.com/trixi-framework/Trixi.jl/pull/1901/commits/b88f12841b3156045ff535053c255fd1eaf0d02e).